src/test/java/com/mysql/cj/jdbc/ha/plugins/ReadWriteSplittingPluginTest.java

```diff
@@ -0,0 +1,636 @@
+package com.mysql.cj.jdbc.ha.plugins;
+
+import com.mysql.cj.conf.ConnectionUrl;
+import com.mysql.cj.conf.HostInfo;
+import com.mysql.cj.conf.PropertySet;
+import com.mysql.cj.exceptions.MysqlErrorNumbers;
+import com.mysql.cj.jdbc.ConnectionImpl;
+import com.mysql.cj.jdbc.JdbcPropertySet;
+import com.mysql.cj.jdbc.JdbcPropertySetImpl;
+import com.mysql.cj.log.Log;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Properties;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.AdditionalMatchers.not;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+public class ReadWriteSplittingPluginTest {
+  private static final int WRITER_INDEX = 0;
+
+  @Test
+  public void testHostInfoStored() throws SQLException {
+    String url = "jdbc:mysql:aws://writer,reader1,reader2/test?" +
+        "connectionPluginFactories=com.mysql.cj.jdbc.ha.plugins.ReadWriteSplittingPluginFactory";
+    ConnectionUrl connUrl = ConnectionUrl.getConnectionUrlInstance(url, new Properties());
+
+    ICurrentConnectionProvider mockCurrentConnectionProvider = Mockito.mock(ICurrentConnectionProvider.class);
+    PropertySet mockPropertySet = Mockito.mock(PropertySet.class);
```

Comment:
https://github.com/awslabs/aws-mysql-jdbc/tree/main/src/test/java/com/mysql/cj/jdbc/ha/plugins
Is it possible to use a similar mocking style to the other plugin tests?

Reply:
https://frontbackend.com/java/difference-between-mockito-mock-method-and-the-mock-annotation